lib/src/builders/method.dart

```diff
@@ -347,7 +395,27 @@ class _MethodBuilderImpl extends Object
         _property = property;
 
   @override
-  AstNode buildAst([Scope scope]) => buildFunction(scope);
+  AstNode buildAst([Scope scope]) {
+    if (_name != null) {
```

Comment:
More of a personal style, but I like to use => and ternary operator here: 
buildAst() => _name != null ? buildFunction() : buildExpression();